src/ArrayOfArraysView.hpp

```diff
@@ -878,11 +917,15 @@ class ArrayOfArraysView
 
     forEachArg( [this, begin, end] ( auto & buffer )
     {
-      for( INDEX_TYPE i = begin; i < end; ++i )
+      if( !std::is_trivially_destructible< decltype( buffer[ 0 ] ) >::value )
```

Comment:
This was another bug.

Reply:
What was the consequence of this bug?

Reply:
A segfault when calling this method when the pointers pointed to device memory. So if you had 
```c++
ArrayOfArrays< int > x;
x.move( chai::GPU )
```
Then `x`'s buffers would be pointing to gpu memory and the call to `arrayManipulation::destroy` will called with device pointers.